ideascube/forms.py

```diff
@@ -52,3 +60,37 @@ def save(self, *args, **kwargs):
         user.is_staff = True
         user.save()
         return user
+
+
+class UserImportForm(forms.Form):
+    source = forms.FileField(required=True)
+
+    def save(self):
+        source = self.cleaned_data['source']
+        users = []
+        errors = []
+        rows = csv.DictReader(source.read().decode().splitlines())
+        for idx, row in enumerate(rows):
+            serial = row.get('serial')
+            if not serial:
+                continue
+            data = {}
+            for field in User._meta.fields:
+                try:
+                    data[field.name] = row[field.name]
+                except KeyError:
+                    continue
+                if isinstance(field, models.CharField):
+                    data[field.name] = to_unicode(data[field.name])
+
+            instance = User.objects.filter(serial=serial).first()
+            form = UserForm(data=data, instance=instance)
+            if form.is_valid():
+                user = form.save()
+                users.append(user)
+            else:
+                reason = ', '.join('{}: {}'.format(k, v.as_text())
+                                   for k, v in form.errors.items())
+                errors.append(_('Invalid row at line {id}: {reason}'.format(
+                    id=idx + 1, reason=reason)))
```

Comment:
This would all make more sense to me if you were passing the `row` directly as data to the `UserForm`, since it handles validating it anyway.

It would also give you error messages for when a row doesn't contain the `serial`, for example (which you're silently ignoring above).

In short, it would make this function shorter and better.